sys/shell/shell.c

```diff
@@ -62,6 +63,10 @@ XFA_INIT_CONST(shell_command_t*, shell_commands_xfa);
 
 #define PARSE_ESCAPE_MASK 0x4;
 
+extern void shell_lock_checkpoint(char *line_buf, int len);
+extern bool shell_lock_is_locked(void);
```

Comment:
If you are doing that you might as well

```suggestion
#if IS_USED(MODULE_SHELL_LOCK)
extern void shell_lock_checkpoint(char *line_buf, int len);
extern bool shell_lock_is_locked(void);
#else
static inline void shell_lock_checkpoint(char *line_buf, int len)
{
    (void)line_buf;
    (void)len;
}

static inline bool shell_lock_is_locked(void)
{
    return false;
}
#endif 
```

and avoid cluttering the code further down with `if (IS_USED(…))` blocks